core/modules/filter-tracker.js

```diff
@@ -0,0 +1,87 @@
+/*\
+title: $:/core/modules/filter-tracker.js
+type: application/javascript
+module-type: global
+
+Class to track the results of a filter string
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+function FilterTracker(wiki) {
+	this.wiki = wiki;
+	this.trackers = [];
+	this.wiki.addEventListener("change",this.handleChangeEvent.bind(this));
+}
+
+FilterTracker.prototype.handleChangeEvent = function(changes) {
+	this.processTrackers();
+	this.processChanges(changes);
+};
+
+/*
+Add a tracker to the filter tracker
+filterString: the filter string to track
+fnEnter: function to call when a title enters the filter results. Called even if the tiddler does not actually exist. Called as (title), and should return a truthy value that is stored in the tracker as the "enterValue"
+fnLeave: function to call when a title leaves the filter results. Called as (title,enterValue)
+fnChange: function to call when a tiddler changes in the filter results. Only called for filter results that identify a tiddler or shadow tiddler. Called as (title,enterValue), and may optionally return a replacement enterValue
+*/
+FilterTracker.prototype.track = function(filterString,fnEnter,fnLeave,fnChange) {
+	// Add the tracker details
+	var index = this.trackers.length;
+	this.trackers.push({
+		filterString: filterString,
+		fnEnter: fnEnter,
+		fnLeave: fnLeave,
+		fnChange: fnChange,
```

Comment:
I am not 100% sure, but there seem to be no tests if the `fn*` parameters are undefined.